llvm/lib/Target/BPF/BPFAsmPrinter.cpp

```diff
@@ -59,7 +59,9 @@ bool BPFAsmPrinter::doInitialization(Module &M) {
   AsmPrinter::doInitialization(M);
 
   // Only emit BTF when debuginfo available.
-  if (MAI->doesSupportDebugInformation() && !M.debug_compile_units().empty()) {
+  // Unsupported for Solana: https://github.com/solana-labs/llvm-project/issues/37
+  if (MAI->doesSupportDebugInformation() && !M.debug_compile_units().empty() &&
+      TM.getTargetTriple().getVendor() != Triple::VendorType::Solana) {
```

Comment:
Checking for `VendorType::Solana` isn't enough I think. We need to disable BTF also when compiling rust code with `--target=bpfel-unknown-unknown` - which is what `cargo build-bpf` uses today - which corresponds to triple=bpfel-unknown-unknown target-features=solana.

See https://github.com/solana-labs/llvm-project/blob/05f07462070dd7f135a1d435149c32f7e343fd47/llvm/lib/Target/BPF/MCTargetDesc/BPFAsmBackend.cpp#L28